src/services/webhdfs/backend.rs

```diff
@@ -154,6 +154,19 @@ impl WebhdfsBuilder {
     /// prepend `http://` to it.
     pub fn endpoint(&mut self, endpoint: &str) -> &mut Self {
         if !endpoint.is_empty() {
+            // check scheme
+            let endpoint = {
+                let uri: Vec<_> = endpoint.split("://").collect();
```

Comment:
How about:

https://github.com/datafuselabs/opendal/blob/fbb74e16f2061f4bad6961f5a2a89a61dee75890/src/services/s3/backend.rs#L691-L701

Reply:
In the future, I plan to use `http::uri` to prase into `Uri` struct and do check over it.

Reply:
I'll move the check to `build` phase.